spring-security-mvc-boot/src/main/java/org/baeldung/custom/web/MainController.java

```diff
@@ -52,7 +55,17 @@ public Foo findFooByName(@RequestParam final String name) {
     @RequestMapping(method = RequestMethod.GET, value = "/organizations/{id}")
     @ResponseBody
     public Organization findOrgById(@PathVariable final long id) {
-        return organizationRepository.findById(id).orElse(null);
+        return organizationRepository.findById(id)
+            .orElse(null);
+    }
+
+    @PreAuthorize("hasPermission(#id, 'Foo', 'read')")
+    @RequestMapping(method = RequestMethod.GET, value = "/user")
```

Comment:
While we are busy simplifying, let's please change these to:

```suggestion
    @GetMapping("/user")
```